### DISCUSSION TITLE
Quote Widget / Quote Module Request

### DISCUSSION THREAD

**Discussion by srcarney** [templaza/astroid-framework#917] [Ideas]
I am building my first website using Astroid Framework and the folks I have building it for LOVE the Quote Article Type. However, they also wish we could **Place a Quote Anywhere**. That made me think that a Quote widget would be really helpful. And maybe a module since it would be good to be able to place a Quote module anywhere.

**Comment by sonvnn** [comment]
You can try testimonial widget. It's available for testing

**Reply by srcarney** [reply]
Thanks for that great idea.

**Comment by srcarney** [comment]
I will try it to see. I wish that both the Quotes and the Testimonials could also somehow be placed in the midst of an article, or anywhere else for that matter. But I realize why something that is added as a section in a layout won't do that. Basically I am wanting modules that do some of the things that are being built into this framework. Anyhow, I love this framework. AND, I would love modules that do some of the good things that are built into this framework.

**Reply by sonvnn** [reply]
Edit section -> Container -> Select without Container

**Reply by srcarney** [reply]
Thank you Sonny. I am assuming you are leading me to a solution regarding my request. I set up a Testimonials Sublayout called Pullquotes and saved it with the section layout of "Without Container".  Now I am trying to figure out where I need to place this within the sites layout in order to place a pullquote in the midst of an article. Is that possible. I figured maybe the "Without Container" option would maybe allow me to create a Position name that this could be published into?

**Reply by srcarney** [reply]
![image](https://github.com/user-attachments/assets/1f78be8a-f112-4507-8cc0-7a77b34aab60)

**Reply by srcarney** [reply]
You can see I created a Pullquote sublayout and took a wild guess about where to place it. I am now trying to figure out how to find a way in my Joomla back-end so that I can create a Pullquote and place it in an article.

**Comment by srcarney** [comment]
This one is sure baffling me. I will stop trying and go watch all the videos and documentation again. I am looking forward to understanding how this works. :-)

**Reply by srcarney** [reply]
I have a feeling that I am misunderstanding this Testimonials widget. I was hoping it would work like the Quote in the Quote article type where when you start a new article you can then add a Quote to the article by filling in all the Quote Fields.  What I am thinking now is that the Testimonials Widget will let me put in a bunch of Quotes and then somehow I will be able to select one of them or have them rotate like a slideshow.  I was sort of hoping something more like a module where I could add a Quote anywhere on the fly so that each Article could have one quote embedded.

**Comment by sonvnn** [comment]
@srcarney Sorry for the wait. Here's a guide on how to integrate Testimonial into your articles.
1. To use Testimonial Widget in Article. You have to create a Article Layout. `Go to Template options > Article Layout Builder.` You can learn more about Article Layout Builder https://docs.astroidframe.work/article-blog/article-layout-builder
2. Create/Edit your article layout and add your Testimonial Widget. Please check an example from Two below 
![Screen Shot 2024-12-17 at 08 21 33](https://github.com/user-attachments/assets/e5533389-7065-4b6a-b09d-a96a8b82d79d)
3. In Testimonial widget options you can configure default values which will display on all articles
![Screen Shot 2024-12-17 at 08 27 32](https://github.com/user-attachments/assets/4ab3ceb9-2818-4a31-9202-2c0f6fa76a86)

### Edit testimonial in Article
To use Article Layout you have to assign it to a Joomla category. In Two template, We use `Photography` category to display it.
![Screen Shot 2024-12-17 at 08 32 44](https://github.com/user-attachments/assets/bd5aa697-6ad9-45c9-8b3a-a9d54c8909dc)

To edit testimonial in each article. You can go to `Administrator -> Content -> New/Edit an Article -> Select Widgets tab`. 
![Screen Shot 2024-12-17 at 08 29 41](https://github.com/user-attachments/assets/3fb671d0-399f-4ce9-ba62-920d956c77ad)

Go to front-end you can have a result like this https://demo.astroidframe.work/article-layouts/photography-example/love-at-first-snap

You can download the Quickstart of Two here to see in detail how Testimonial works on Article Layout https://github.com/templaza/astroid-framework/releases/tag/v3.1.9

Please let's me know if you need further assistant.

Best Regards,
Sonny

**Comment by srcarney** [comment]
Ah... thank you so much. The piece that I was missing was that I would have to create an Article Layout that included the Testimonials widget. 

That is so great. I realize then that I could get the real nice layouts. It also looks to me like I would have to decide exactly where to  position the Testimonial and that it is not going to actually display in the middle of the full text somewhere as if I was using a load position module to load a module. 

Or is that possible because of selecting the option to not be Without Container?

I would like to place the quote somewhere in the midst of the article's full text but it may not  be the exact same place each time. 

I was just looking again at your Quotes Article Type option before I saw that you had replied here and I was thinking how cool it would be if instead of having only three positions to place the quote, if there were another one or two positions that could manually be entered somewhere within the text of the full text. Or anywhere else for that matter. Also another thing that would be cool  if if more than one color of background was available..

**Comment by srcarney** [comment]
I have a Quickstart of Astroid 3.1.7 but am thinking that the testimonials were not in that Quickstart. I will download the new 3.1.9 and check it out.

**Comment by srcarney** [comment]
OK. I just installed the 3.1.9 Quiickstart and now I see how the Testimonials work. Basically for every testimony/quote or set of testimonies/quotes, I will need to create a Article Layout and assign it to a specific category. This makes is possible for me to add testimonies/quotes but doesn't give me the ability for my authors to insert a quote unless they or I have also created the Articles Layout. 

I much prefer the way you are doing it in the Article Type Options where the user can select the option to make this a Quote. Then they just put in the Quote Text, the Quote Author, and select the Position and the beautiful quote displays. The people I am making the site love the look of this but wish that those Quotes could be placed in the middle of an article using some sort of a tag. 

In your Quickstart a beautiful Quote of the type they are loving is found at https://two319qs.freesite.host/index.php/article-layouts/photography-example/shutter-up-and-smile

Two questions:  

1) How would the editor move that quote down to somewhere of their choosing in the middle of the article?  Is that possible? 
2) Can you think of any other way where Quotes and Testimonies, etc. could be added on the fly by an editor without getting them involved in creating Article layouts? Or is using the built in Blockquote tag our only option in that case?

Thanks again!!!

**Reply by sonvnn** [reply]
You will see the Article Content block on Article Layout Builder. Let's try to move it to anywhere on the Layout. You will see the Quote block move with it on the front-end.

**Reply by sonvnn** [reply]
![Screen Shot 2024-12-20 at 15 25 14](https://github.com/user-attachments/assets/f26876ab-3ff4-47ea-b57b-6b63b47b2b7a)

**Comment by srcarney** [comment]
That is very cool. This widget system is powerful. In the example you gave me I still do not see how I could place a Quote within an Article Quote Type article in the middle of an Article. It seems I can put it above the article or below the article but that there is no way to create a position in the middle of an article. 

On the following page I can place the quote above the article:
![image](https://github.com/user-attachments/assets/203ff1f3-1ef9-4ffc-a188-b2d99a869ed8)

or below the article:
![image](https://github.com/user-attachments/assets/df80d877-6137-4998-92e1-59630031c98f)

But, what I keep trying to suggest as a really great possibility would be to develop the ability to place the quote between paragraphs in the middle of the article.
 
![Screenshot 2024-12-20 122150](https://github.com/user-attachments/assets/fbbc8c8c-394c-4099-b0b4-b2c65ba16482)

**Reply by srcarney** [reply]
I am sorry. The third image was supposed to have a line in between the second and the third paragraph stating that I would like to see the beautifully formatted quote there.

**Reply by srcarney** [reply]
![Screenshot 2024-12-20 122340](https://github.com/user-attachments/assets/fc98b8e8-604b-42c7-ba18-0113cc3a634f)

**Reply by srcarney** [reply]
BTW, I am putting this in as a request for something to consider. But if it can't be done, I will still be a strong supporter of the Astroid Framework.

**Reply by sonvnn** [reply]
You can use introtext & fulltext to split whole text. Let's try with it.

**Reply by srcarney** [reply]
That is definitely a good possibility.  I am actually using Intro Text on this site on every article to set up what is displayed in the Category and Subcategory pages. Then when a person clicks, they are taken to the full text. So this idea won't work. 

Anyhow, I have consumed too much of your time on this. I understand that what I am asking for is not going to happen and probably for good reason. So, I am going to discontinue this discussion so that you can return to your important work of making Astroid great. 

Thanks again,  Sean Carney

**Reply by sonvnn** [reply]
Can you make a draft of your idea by image? I can follow it and give you a solution.

**Comment by srcarney** [comment]
The image below shows the Three Options for Position in Article Type Options of the Quote Type. You will notice there are three options:

Above Content
Before Content
After Content

This is standard Joomla behavior. 

I would love for there to be a fourth option which is where you could consider extending the system. But it would require somehow adding module position and that could be inserted into the article. 

I think it could be called "Inside Content"

Then there would be four options. One of them:

Above Content
Before Content
Inside Content
After Content

There are probably a few options for telling the system where to insert the Quote image and text. Either using a tag to place the position between the paragraphs where the user wants it to display. or... maybe setting a number for the quantity of paragraphs to display before displaying the image. I am sure there are other ideas too. 

![image](https://github.com/user-attachments/assets/a48d0483-d665-42e1-b823-a29d03728bbd)

On the following image you will see an area where I just put a Joomla blockquote. But, my preference would be to be able to display the quote that is nicely formatted from the template into that space. 

![image](https://github.com/user-attachments/assets/600e97e2-104e-464d-9c31-b4c6f0a2d4cc)

This solution would only allow the image to be moved to the middle of an article if we are using the Quote Type Option. 

What would be REALLY cool and I suspect people would use it is if there was a way in Astroid to add a quote like this into the middle of any type of Astroid / Joomla article. I am not sure how you would do that. But, it would be something much nicer than the joomla blockquotes. 

These are called Pull Quotes and are very common in magazines where the Pull Quote is actually in the article already but it is highlighted in a beautiful way as well somewhere on the page.

**Reply by sonvnn** [reply]
Please create a ticket and provide your admin account. I would like to make a test for this.

**Comment by sonvnn** [comment]
@srcarney I reproduce my process with the "[Shutter Up and Smile](https://demo.astroidframe.work/article-layouts/photography-example/shutter-up-and-smile)" article so you can understand how Article Layout Builder works.

1. Create an article layout on template options
![article_layout_builder](https://github.com/user-attachments/assets/f2750f77-93a9-461c-acf7-870781491385)
2. Go to Content -> Articles -> Edit your article that assign to category of Article Layout -> Widgets Tab
![editcontent](https://github.com/user-attachments/assets/a6019960-e164-4ed7-92ae-009feaca3be3)
3. Input data of Testimonials and Article Quote Type
![editmodal](https://github.com/user-attachments/assets/2b0dd209-50b4-41b0-bff5-291e01506f81)
![Screen Shot 2024-12-25 at 10 26 41](https://github.com/user-attachments/assets/73ecbfa4-ddaf-44d7-a198-e1c2af856a7f)
4. Go to front-end and check the result
![screenshot](https://github.com/user-attachments/assets/321184ab-94d0-44ea-85ff-c633e8701050)

I hope it’s clear enough for you to understand how Article Layout Builder works. You don’t need to use any load-module plugin here. The widget system will help you do whatever you want.

Best Regards,
Sonny

**Comment by srcarney** [comment]
Thank you so much. I have gotten much further along. I am hoping I can you this. 

Here are the two things I am having problems with: 

1) I am unable to place the quote in the middle of the Full Text. It has to be either above or below

2) I am unable to give the quote a background image since you did that in the Quickstart by putting a background image behind a Section. For example here on https://travelami.cloudaccess.host/index.php/about-us

**Reply by srcarney** [reply]
Here is my page I am experimenting on: https://shalomd1.jdevcloud.com/scripture/layout-tests/testing-the-quote-article-type-length

I opened a ticket at templaza.com with super user access instructions. So far I have to say that this appears to be much harder than just placing a module into the document. But... it certainly has the potential to be much better than using modules. I must still be missing something.

**Reply by srcarney** [reply]
I can add the images as long as I am not using settings to Card. So, I have sort of solved that. I just with that I could place the quote somewhere besides the end of the article or the beginning. However, I will be using this feature even if that is not possible. I just have to change the way I want to use it. 

I do have another question. I am setting up my default settings in the Layout that I created. Is there any way to open the Testimonials I have opened on a few pages and either have them revert to the latest default template setting, or delete them and have the new testimonial use the default settings. Once I have he default settings, I will let the editors add the quotes.

**Reply by srcarney** [reply]
Also, for some reason the background colors that I am setting up are not being displayed. Here are four urls where I have set up some Testimonial quotes at the bottom of the pages:

https://shalomd1.jdevcloud.com/scripture/layout-tests/testing-the-quote-article-type-length

https://shalomd1.jdevcloud.com/scripture/layout-tests/testing-the-review-article-type-length

https://shalomd1.jdevcloud.com/scripture/layout-tests/testing-the-audio-article-type-length

https://shalomd1.jdevcloud.com/scripture/layout-tests/testing-the-gallery-article-type-length

**Comment by srcarney** [comment]
How can I remove all of the test quotes that I have added to my Layout Test Articles?

And, how can I make the existing test quotes inherit the latest Testimonial widget settings?

I will want to make some changes to the Testimonials widget settings but can't remove or modify the quotes that I have already added.

**Reply by sonvnn** [reply]
To remove testimonial widget. Go to Edit your article >> Widgets tab >> Select "Testimonial Widget" >> Remove all Content. Save it.

**Reply by srcarney** [reply]
Thanks. Nice and simple. I was also expecting that when I removed the content the formatting would also revert back to whatever I set as default. But that is not the case. So, I have a suggestion but don't know how hard it would be to implement. I would like to have a button that would remove all content (we kind of have that already) and also remove the formatting and replace it with the default formatting options that we set up coming from the Widget settings. :-)
